python_transport/wirepas_gateway/protocol/mqtt_wrapper.py

```diff
@@ -290,15 +292,20 @@ def fileno(self):
         return self._getsocket.fileno()
 
     def put(self, item, block=True, timeout=None):
-        # Insert item in queue
-        super().put(item, block, timeout)
-        # Send 1 byte on socket to signal select
-        self._putsocket.send(b"x")
+        with self._lock:
+            if self.empty():
```

Comment:
From the queue documentation:

_Return True if the queue is empty, False otherwise (not reliable!)._
_This method is likely to be removed at some point._

Any idea why it is not reliable and what could be the consequences?

As an alternative, I would keep the size of the queue as an attribute of this child class. We would have the control of it instead of relaying on the empty() implementation.

Reply:
My understanding is that the "not reliable" is a warning for multi-threaded issues.
The doc (https://docs.python.org/3.8/library/queue.html) says:
Return True if the queue is empty, False otherwise. If empty() returns True it doesn’t guarantee that a subsequent call to put() will not block. Similarly, if empty() returns False it doesn’t guarantee that a subsequent call to get() will not block.

And there is not mention of a deprecation.

It's hard to believe that a Queue implementation is not able to provide the number of elements...